projects/packages/videopress/src/class-jwt-token-bridge.php

```diff
@@ -30,8 +30,7 @@ public static function init() {
 			return;
 		}
 
-		// Expose the VideoPress token to the Block Editor context, including the front-end editor.
-		add_action( 'enqueue_block_assets', array( __CLASS__, 'enqueue_jwt_token_bridge' ), 1 );
```

Comment:
I think this was necessary since we don't know ahead of time if there will be a video block added to a new post.

But if you have some way of detecting that and enqueuing it afterwards then that might be OK?

Reply:
First and just in case, we need to keep in mind that we are enqueuing the token-bridge file in the block editor content line below via the `enqueue_block_editor_assets` action. It guarantees to deal with private videos in this context (block editor)

The `enqueue_block_assets` action [fires after enqueuing block assets for both editor and front-end](https://developer.wordpress.org/reference/hooks/enqueue_block_assets/).

Enqueuing the token bridge by hooking to `enqueue_block_assets` is not optimal (IMHO) since it could be possible to create a VideoPress video instance using other methods, like a shortcode or a simple URL?
Ultimately, the app will filter the VideoPress URL request to the oEmbed markup to serve it to the client.

I tested with a shortcode:
<img width="724" alt="image" src="https://user-images.githubusercontent.com/77539/232849205-4ba35f2e-e42b-40f8-a415-11233ce973a6.png">

With a raw URL:
<img width="734" alt="image" src="https://user-images.githubusercontent.com/77539/232849140-c19b987e-c1bc-4943-ba8b-82306753a54d.png">

VideoPress embed block
<img width="730" alt="image" src="https://user-images.githubusercontent.com/77539/232849639-e761d160-13f0-4877-8263-abe7c274eeff.png">

Extended Core Video block (v5)

<img width="726" alt="image" src="https://user-images.githubusercontent.com/77539/232849881-b02fe792-f83e-4f05-aaf3-3940cd2b8bb2.png">

...and the token bridge is always enqueued via the `embed_oembed_html` hook.

If we don't remove that line, the script is always enqueued. It doesn't matter if the site has or hasn't a VideoPess video block, a shortcode, or whatever.